### DISCUSSION TITLE
converting to float32 before torch.mean()

### DISCUSSION THREAD

**Discussion by arashmehrabiparsa** [mrdbourke/pytorch-deep-learning#704] [General]
how do we convert from different datatypes to float32 in order to perform the torch.mean() https://youtu.be/Z_ikDlimN6A?si=2NSR412c0y1Vpjpl&t=10073? , it seems it converts automatically when specifying x.type(torch.float32) right?

**Comment by shisirkha** [comment]
Yeah, it like specifying `x.type(torch.float)`(Note pure float means float32) but you need to do this if you want to make it permanent.
```
x = x.type(torch.float)
```
